frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/useServingRuntimeConfig.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { ServingRuntimeKind } from 'k8sTypes';
+import { fetchServingRuntimeConfig } from 'api';
+import { useDashboardNamespace } from 'redux/selectors';
+
+const useServingRuntimeConfig = (
+  open: boolean,
+): [
+  servingRuntimeConfig: ServingRuntimeKind | null,
+  loaded: boolean,
+  loadError: Error | undefined,
+] => {
+  const { dashboardNamespace } = useDashboardNamespace();
+  const [servingRuntimeConfig, setServingRuntimeConfig] = React.useState<ServingRuntimeKind | null>(
+    null,
+  );
+  const [loaded, setLoaded] = React.useState(false);
+  const [loadError, setLoadError] = React.useState<Error | undefined>(undefined);
+
+  const resetFetch = () => {
+    setLoadError(undefined);
+    setServingRuntimeConfig(null);
+    setLoaded(false);
+  };
+
+  const getServingRuntimeConfig = React.useCallback(() => {
+    resetFetch();
```

Comment:
Clearing values like this is problematic -- since you'll lose what you currently have. Is this needed -- why is the stale value so bad?

Reply:
I guess i can update the error/result in the try/catch, I did this to empty the error status in case the fetch is successful, but might be better if I just do that in the `then` clause.

Reply:
Yeah -- definitely just update the error to nothing in the `.then` -- we have a tech debt ticket to make fetching easier where this is not a conversation topic as we'll have a generic fetcher that does all this.

Reply:
ok, nice, updating it!

Reply:
done